core/src/subgraph/instance.rs

```diff
@@ -168,13 +168,27 @@ where
                     .map(Arc::new)
                     .context("Found no transaction for event")?;
                 let matching_hosts = hosts.iter().filter(|host| host.matches_log(&log));
+                let hosts_count = matching_hosts.clone().count();
```

Comment:
Iterators are a good candidate for CheapClone. This reads like we're coming the runtime hosts, but really, we're just cleaning the iterator. CheapClone would make this more obvious. Our a comment.

Reply:
Yeah, it's just cloning the iterator and then it's dropped immediately after counting.  I'll cheap clone and add a comment to make it clear.